src/Lean/Meta/InferType.lean

```diff
@@ -203,25 +203,33 @@ private def isAlwaysZero : Level → Bool
   | .max u v    => isAlwaysZero u && isAlwaysZero v
   | .imax _ u   => isAlwaysZero u
 
+private def isAlwaysZeroQuick : Level → LBool
```

Comment:
This isn't a "quick" approximation, it's a actually a decision procedure, or am I missing something?

Reply:
That's true. I don't know if we have a naming scheme here, `isAlwayZeroNoInst`?

Reply:
I'd go for `isAlwaysZero`.  I think `Always` already implies `NoInst`.  We have a very similar function called `isNeverZero` already.

Reply:
We still have the original `isAlwaysZero` above though. But since that one is now used only in the non-quick `isProp` branch, we could probably replace it with this version+`== .true` for insignificant overhead.

Reply:
Yeah, I think replacing `isAlwaysZero` by the `LBool` version is probably the cleanest approach.  We should also mark `LBool.and` as `@[macro_inline]`.

Unfortunately `Level.isZero` is already taken, or we could define `isZero : Level → LBool` and `isAlwaysZero l := l.isZero == .true` and `isNeverZero l := l.isZeo == .false`.

Reply:
I think it's fine to leave it as a single `private` here as long as we don't need it elsewhere